File: Inventory/Domain/Product.cs
using System;

namespace Inventory.Domain
{
    public class Product
    {
        public Guid ProductId { get; set; }
        public Category Category { get; set; }
        public string Name { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public decimal Cost { get; set; }
        public Uri ImageUri { get; set; }
    }

    public enum Category
    {
        Physical,
        EventTicket,
        Service
    }
}

File: angular/src/app/products/product.ts
import {IProduct} from "./products.service";

export class Product implements IProduct {
    productId: string = '';
    category: string = '';
    cost: number = 0.00;
    imageUri: string = '';
    name: string = '';
    price: number = 0.00;
    quantity: number = 0;
}
